actions/linting/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Linting
+description: Runs pre-commit over the repo
+
+inputs:
+  pre-commit-config:
+    description: The pre-commit config file of the current repo
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout source
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
+
+    - name: Cache pre-commit
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pre-commit
+        key: pre-commit-${{ hashFiles(inputs.pre-commit-config) }}
+
+    - name: Set up python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+
+    - name: Install dependencies
+      shell: bash
+      run: python -m pip install pre-commit
+
+    - name: Run pre-commit
+      shell: bash
```

Comment:
Looks like `shell` is required for `composite actions` https://stackoverflow.com/a/71042698/7359333